rustv1/examples/s3/src/bin/select-object-content.rs

```diff
@@ -49,14 +51,25 @@ struct Record {
     phone_number: String,
     city: String,
     occupation: String,
+    description: String,
+}
+
+fn is_valid_json(data: impl AsRef<str>) -> bool {
+    serde_json::from_str::<IgnoredAny>(data.as_ref()).is_ok()
 }
 
 // Get object content.
 // snippet-start:[s3.rust.select-object-content]
-async fn get_content(client: &Client, bucket: &str, object: &str, name: &str) -> Result<(), Error> {
-    let mut person: String = "SELECT * FROM s3object s WHERE s.\"Name\" = '".to_owned();
-    person.push_str(name);
-    person.push('\'');
+async fn get_content(
+    client: &Client,
+    bucket: &str,
+    object: &str,
+    name: &str,
+) -> Result<(), anyhow::Error> {
+    // To escape a single quote, use two single quotes.
+    let name = name.replace("'", "''");
+    let person: String = format!("SELECT * FROM s3object s where s.Name like '{name}%'");
```

Comment:
How safe is string interpolation in S3 select?

Reply:
this is probably necessary, but perhaps not sufficient for all cases. However, I found that most attempts at injection just caused S3's parser to barf. If you were doing IRL handing of untrusted inputs in S3 select, you may want to do something like constraining your input to alpha only, e.g.